File: QViz.NET/Periscope.QViz/JSON/TestStepResult.cs
using System;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for GUI Test Step Result Details
	/// </summary>
	public class TestStepResult
	{
		/// <summary>
		/// Identifier of the Test Case Step
		/// </summary>
		public string testCaseStepId;
		
		/// <summary>
		/// Actual result of the Test Case Step execution
		/// </summary>
		public string actualResult;
		
		/// <summary>
		/// URL of the Screen shot taken when exception occurred during the execution of the Test Case Step
		/// </summary>
		public string screenshotURL;
		
		/// <summary>
		/// Status of the Test Case Step execution (Pass, Fail, Broken)
		/// </summary>
		public string status;
		
		/// <summary>
		/// Error message from the execution of the Test Case Step
		/// </summary>
		public string error;
		
		/// <summary>
		/// Date and Time on which the execution of the Test Case Step was started
		/// </summary>
		public DateTime executionStartTime;
		
		/// <summary>
		/// Date nad Time on which the execution of the Test Case Step was finished
		/// </summary>
		public DateTime executionEndTime;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public TestStepResult()
		{
			this.testCaseStepId = "";
			this.actualResult = "";
			this.screenshotURL = "";
			this.status = "";
			this.error = "";
			this.executionStartTime = DateTime.Now;
			this.executionEndTime = DateTime.Now;
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/TestStepResult.java
package com.periscope.qviz.json;

import java.util.Date;

/**
 * JSON Model for GUI Test Step Result Details
 */
public class TestStepResult {

	/**
	 * Identifier of the Test Case Step
	 */
	public String testCaseStepId;

	/**
	 * Actual result of the Test Case Step execution
	 */
	public String actualResult;

	/**
	 * URL of the Screen shot taken when exception occurred during the execution of the Test Case Step
	 */
	public String screenshotURL;

	/**
	 * Status of the Test Case Step execution (Pass, Fail, Broken)
	 */
	public String status;

	/**
	 * Error message from the execution of the Test Case Step
	 */
	public String error;

	/**
	 * Date and Time on which the execution of the Test Case Step was started
	 */
	public Date executionStartTime;

	/**
	 * Date nad Time on which the execution of the Test Case Step was finished
	 */
	public Date executionEndTime;

	/**
	 * Default Constructor
	 */
	public TestStepResult() {
		this.testCaseStepId = "";
		this.actualResult = "";
		this.screenshotURL = "";
		this.status = "";
		this.error = "";
		this.executionStartTime = null;
		this.executionEndTime = null;
	}

}
